src/megaapi_impl.cpp

```diff
@@ -18186,16 +18186,37 @@ void MegaApiImpl::sendPendingRequests()
 {
     int nextTag = 0;
 
-    while(true)
+    SdkMutexGuard g(sdkMutex);
+    
+    // For multiple consecutive requests of the same type (eg. remove transfer) this committer will put all the database activity into a single commit
+    DBTableTransactionCommitter committer(client->tctable);
+    int lastRequestType = -1;
+    int lastRequestConsecutive = 0;
+
+
+    while(MegaRequestPrivate *request = requestQueue.front())
     {
-        sdkMutex.lock();
 
-        MegaRequestPrivate *request = requestQueue.pop();
-        if (!request)
+        // also we avoid yielding for consecutive transaction cancel operations (we used to yeild every time, but we need to keep the sdkMutex lock while the database transaction is ongoing)
+        if ((lastRequestType == -1 || lastRequestType == request->getType()) && lastRequestConsecutive < 1024)
```

Comment:
Why do we check the `lastRequestType` here? :thinking: 
Perhaps it's cleaner to reset the value when we break the loop.

Reply:
The change from the comment below will fix this up - we won't break the loop at all, but rather it finishes when we run out of items on the request queue - so this line is specifically to decide if we need to commit and start a new database transaction (but lazily, so we only actually begin the transaction if any database work gets started).   And we do that if we are processing a different request type to the last loop iteration (if there was one), or if we have done 1024 of the same request type in a row.  thanks!

Reply:
Oh, I see now. My bad, I didn't notice, at first, that a different type of request should commit the ongoing transaction. All good :)